sycl/include/CL/sycl/half_type.hpp

```diff
@@ -27,11 +27,122 @@
 #define __SYCL_CONSTEXPR_ON_DEVICE
 #endif
 
+#ifndef __has_builtin
+#define __has_builtin(x) 0
+#endif
+
+#if !__has_builtin(__builtin_expect)
+#define __builtin_expect(a, b) (a)
+#endif
```

Comment:
Note: we already have those macro available in `defines_elementary.hpp` - probably it makes sense to re-use them from there.

Reply:
Define of __has_builtin was removed